.github/workflows/syntax-checks.yml

```diff
@@ -1,6 +1,13 @@
 name: Syntax Checks
 
-on: [push, pull_request]
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
 
```

Comment:
additionally, only for pushes to main branch and for PRs that will merge into main branch, correct?